core/src/operations/mod.rs

```diff
@@ -109,6 +109,7 @@ pub(super) mod opcode_constants {
     pub const OPCODE_JOIN: u8       = 0b0101_0111;
     pub const OPCODE_DYN: u8        = 0b0101_1000;
     pub const OPCODE_RCOMBBASE: u8  = 0b0101_1001;
+    pub const OPCODE_EMIT: u8       = 0b0101_1010;
 
     pub const OPCODE_MRUPDATE: u8   = 0b0110_0000;
     pub const OPCODE_PUSH: u8       = 0b0110_0100;
```

Comment:
Unless it causes some complications, I'd probably still move the `PUSH` opcode to be immediately after the `EMIT` opcode (e.g., be `0b0101_1011`). Not only would this lower the degree of $f_{imm}$ flag, but it would also mean that we can save one operation when computing $f_{imm}$ (i.e., we'd get it naturally in the process of computing `PUSH`/`EMIT` flags.

Reply:
Moved PUSH to `0b0101_1011`.

> this lower the degree of $f_{imm}$ flag,

The degree doesn't change though right? Since now both `f_emit` and `f_push` are degree 5.

Reply:
I think by virtue of having a common prefix and differing only in the first bit, the flags when added together will cause the last bit to cancel out, something like $x * b_0 + x * (1 - b_0) = x$. This is also what happens for `SPLIT` and `LOOP`.
One thing that I just remembered, reductions in degree like these can cause the constraint evaluation logic related to flags to result in the wrong evaluations. This happened a while back and took some time to debug.

Reply:
Yup that makes sense, very interesting. $f_{imm}$ is now set to degree 4